File: ClipboardZanager/Tests/ClipboardZanager.Shared.Tests/Mocks/CloudTokenProviderMock.cs
using System.Collections.Generic;
using ClipboardZanager.Shared.CloudStorage;

namespace ClipboardZanager.Shared.Tests.Mocks
{
    public class CloudTokenProviderMock : ICloudTokenProvider
    {
        public string GetToken(string tokenName)
        {
            if (tokenName == "MyToken")
            {
                return "123456789abc";
            }

            throw new KeyNotFoundException();
        }

        public void SetToken(string tokenName, string value)
        {
            if (tokenName != "MyToken")
            {
                throw new KeyNotFoundException();
            }
        }
    }
}

File: Android/app/src/androidTest/java/com/etiennebaudoux/clipboardzanager/mocks/CloudTokenProviderMock.java
package com.etiennebaudoux.clipboardzanager.mocks;

import com.etiennebaudoux.clipboardzanager.componentmodel.cloudstorage.CloudTokenProvider;

/**
 * Created by ebaud on 2/11/2017.
 */

public class CloudTokenProviderMock implements CloudTokenProvider {
    @Override
    public String getToken(String tokenName) {
        if (tokenName.equals("MyToken"))
        {
            return "123456789abc";
        }

        throw new RuntimeException();
    }

    @Override
    public void setToken(String tokenName, String value) {
        if (!tokenName.equals("MyToken"))
        {
            throw new RuntimeException();
        }
    }
}
